source/cipher_openssl.c

```diff
@@ -457,12 +468,16 @@ int aws_cryptosdk_sig_sign_start_keygen(
     aws_cryptosdk_sig_abort(*pctx);
     *pctx = NULL;
 
-    aws_string_destroy(*pub_key);
-    *pub_key = NULL;
+    if (pub_key) {
+        aws_string_destroy(*pub_key);
+        *pub_key = NULL;
+    }
 
     EC_KEY_free(keypair);
     EC_GROUP_free(group);
 
+    AWS_POSTCONDITION(!*pctx);
+    AWS_POSTCONDITION(!pub_key || !*pub_key);
```

Comment:
Why the disjunction here?

Reply:
A NULL pointer can be passed to the function as `pub_key`.